tests/pytest_loganalyzer/pytest_loganalyzer.py

```diff
@@ -0,0 +1,181 @@
+        if not self._validate_response(result):
+            return {}
+
+        # Download roles/test/files/tools/loganalyzer/loganalyzer.py to the DUT
+        result = self.ansible_host.fetch(dest=SYSLOG_TEMP_FOLDER, src=self.extracted_syslog, flat="yes")
+        if not self._validate_response(result):
+            return {}
+
+        analyzer = ansible_loganalizer(self.run_id, False)
+        analyzer.place_marker_to_file(SYSLOG_TEMP_FOLDER, analyzer.create_end_marker())
+
+        match_messages_regex = re.compile('|'.join(self.match_regex)) if len(self.match_regex) else None
+        ignore_messages_regex = re.compile('|'.join(self.ignore_regex)) if len(self.ignore_regex) else None
+        expect_messages_regex = re.compile('|'.join(self.expect_regex)) if len(self.expect_regex) else None
+
+        analyzer_parse_result = analyzer.analyze_file_list([SYSLOG_TEMP_FOLDER], match_messages_regex, ignore_messages_regex, expect_messages_regex)
+
+        total_match_cnt = 0
+        total_expect_cnt = 0
+        expected_lines_total = []
+        unused_regex_messages = []
+        try:
+            for key, value in analyzer_parse_result.iteritems():
+                matching_lines, expecting_lines = value
+                analyzer_summary["total"]["match"] += len(matching_lines)
+                analyzer_summary["total"]["expected_match"] += len(expecting_lines)
+                analyzer_summary["match_files"][key] = {"match": len(matching_lines), "expected_match": len(expecting_lines)}
+                analyzer_summary["match_messages"][key] = matching_lines
+                analyzer_summary["expect_messages"][key] = expecting_lines
+                expected_lines_total.extend(expecting_lines)
+        except Exception as err:
+            logging.error("{}".format(err))
+            return {}
+
+        # Find unused regex matches
+        for regex in self.expect_regex:
+            for line in expected_lines_total:
+                if re.search(regex, line):
+                    break
+            else:
+                unused_regex_messages.append(regex)
+        analyzer_summary["total"]["expected_missing_match"] = len(unused_regex_messages)
+        analyzer_summary["unused_expected_regexp"] = unused_regex_messages
+
+        return analyzer_summary
+
+    def save_extracted_log(self, dest):
+        """
+        @summary: Download extracted syslog log file to the ansible host.
+
+        @param dest: File path to store downloaded log file.
+        """
+        result = self.ansible_host.fetch(dest=dest, src=self.extracted_syslog, flat="yes")
+        if not self._validate_response(result):
+            return False
+        return True
+
+    def _validate_response(self, response):
+        """
+        @summary: Verify that response obtained by ansible module does not have failed message.
+
+        @param response: Dictionary obtained in result of ansible module execution.
+        """
+        if response.has_key("failed"):

```

Comment:
ansible module is supposed to throw on error (res.is_failed == True). Can you add this in ansible_host.py?